sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/assert-life-time-maximum-full-time-step.ts

```diff
@@ -42,18 +45,44 @@ export class AssertLifeTimeMaximumFullTimeStep implements ECertProcessStep {
     log: ProcessSummary,
   ): Promise<boolean> {
     log.info("Checking life time maximums for BC loans.");
-    for (const disbursementValue of eCertDisbursement.disbursement
-      .disbursementValues) {
-      if (
-        disbursementValue.valueAmount &&
-        disbursementValue.valueType === DisbursementValueType.BCLoan
-      ) {
-        await this.checkLifeTimeMaximumAndAddStudentRestriction(
-          eCertDisbursement,
-          disbursementValue,
+    if (getRestrictionByCode(eCertDisbursement, RestrictionCode.BCLM)) {
+      log.info(
+        `Student already has a ${RestrictionCode.BCLM} restriction, hence skipping the check.`,
+      );
+      return true;
+    }
+    // Check if the BC Loan is present in the awards to be disbursed.
+    const bcLoan = eCertDisbursement.disbursement.disbursementValues.find(
+      (award) => award.valueType === DisbursementValueType.BCLoan,
+    );
+    if (!bcLoan?.valueAmount) {
+      log.info(
+        `${bcLoan.valueCode} award not found or there is no amount to be disbursed, hence skipping the check.`,
+      );
+      return true;
+    }
+    // Check if a new restriction should be created and award adjusted.
+    const newRestrictionCreated =
+      await this.checkLifeTimeMaximumAndAddStudentRestriction(
+        eCertDisbursement,
+        bcLoan,
+        entityManager,
+      );
+    if (newRestrictionCreated) {
+      // If a new restriction was created refresh the active restrictions list.
```

Comment:
Can we have a log.info here? to differentiate that a new restriction was created?

Reply:
We have it already at lines 80 and 84.
```
`New ${RestrictionCode.BCLM} restriction was added to the student account.`
````
````
`No ${RestrictionCode.BCLM} restriction was created at this time.`
````